cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -103,6 +113,12 @@ class ScalarVisitor {
         ss << type->ToString();
         return Status::Invalid(ss.str());
       }
+    } else if (PyList_Check(obj) || PyArray_Check(obj)) {
+      // XXX This code path is used for non-object arrays, which leads
```

Comment:
Can you make a JIRA for this and reference as `// TODO(ARROW-XXX): `?

Reply:
Done.